src/main/java/io/papermc/hangar/service/OrgFactory.java

```diff
@@ -6,48 +6,78 @@
 
 import java.util.Map;
 
 @Component
 public class OrgFactory {
 
     private final HangarDao<OrganizationDao> organizationDao;
+    private final HangarDao<UserDao> userDao;
     private final HangarConfig hangarConfig;
     private final UserService userService;
     private final RoleService roleService;
-
-    /*TEMP*/
-    private final HangarDao<UserDao> userDao;
-    /*END TEMP*/
+    private final NotificationService notificationService;
+    private final RestTemplate restTemplate;
+    private final ObjectMapper mapper;
 
     @Autowired
-    public OrgFactory(HangarDao<OrganizationDao> organizationDao, HangarConfig hangarConfig, UserService userService, RoleService roleService, HangarDao<UserDao> userDao) {
+    public OrgFactory(HangarDao<OrganizationDao> organizationDao, HangarDao<UserDao> userDao, HangarConfig hangarConfig, UserService userService, RoleService roleService, NotificationService notificationService, RestTemplate restTemplate, ObjectMapper mapper) {
         this.organizationDao = organizationDao;
+        this.userDao = userDao;
         this.hangarConfig = hangarConfig;
         this.userService = userService;
         this.roleService = roleService;
-        this.userDao = userDao;
+        this.notificationService = notificationService;
+        this.restTemplate = restTemplate;
+        this.mapper = mapper;
     }
 
     public OrganizationsTable createOrganization(String name, long ownerId, Map<Long, Role> members) {
+        // TODO logging
         String dummyEmail = name.replaceAll("[^a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]", "") + '@' + hangarConfig.org.getDummyEmailDomain();
-        // TODO create dummy user w/email
-        /* TEMPORARY TODO REMOVE */
-        UsersTable usersTable = userDao.get().insert(new UsersTable(name, name, dummyEmail, "I'm an organization", new int[]{ }, false, "en_US"));
-        /* END TEMPORARY */
-        OrganizationsTable org = new OrganizationsTable(name, ownerId, usersTable.getId()); // TODO userId will be from result of dummy user creation
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
+        map.add("api-key", hangarConfig.sso.getApiKey());
+        map.add("username", name);
+        map.add("email", dummyEmail);
+        map.add("verified", Boolean.TRUE.toString());
+        map.add("dummy", Boolean.TRUE.toString());
+        AuthUser authOrgUser;
+        try {
+            authOrgUser = mapper.treeToValue(restTemplate.postForObject(hangarConfig.security.api.getUrl() + "/api/users", new HttpEntity<>(map, headers), ObjectNode.class), AuthUser.class);
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new HangarException("error.org.cannotCreate");
+        }
+        // TODO this will happen via /api/sync_sso, but I have no idea how to get that whole system working with Docker
```

Comment:
Need to verify this but I *think* Ore inserts organizations manually, so the `sso_sync` updates the user instead of creating it (then again, I have no idea what Ore is doing).

Reply:
Yes, Hangar will create the organization and insert it, but during the dummy user creation, hangarauth will create the hangar user table via sync_sso. I can't get that to work atm, so I'm just manually inserting the user.